src/aws-lambda/segment-personalize-destination/segment-personalize-destination.py

```diff
@@ -0,0 +1,128 @@
+        return jData
+    else:
+        myResponse.raise_for_status()
+
+def set_user_traits(user_id, traits):
+    # Sends an identify call to Personas to update a user's traits
+    formatted_url = "{:s}/identify".format(connections_endpoint_url)
+    message = { "traits": traits, "userId": user_id, "type": "identify" }
+    try:
+        response = api_post(formatted_url, connections_source_api_key, message)
+    except HTTPError as error:
+        status = error.response.status_code
+        if status >= 400 and status < 500:
+            logger.error('Segment: 400 error, more than likely you sent an invalid request.')
+        elif status >= 500:
+            logger.error('Segment: There was a server error on the Segment side.')
+
+def lambda_handler(event, context):
+    if not 'personalize_tracking_id' in os.environ:
+        raise Exception('personalize_tracking_id not configured as environment variable')
+    if not 'personalize_campaign_arn' in os.environ:
+        raise Exception('personalize_campaign_arn not configured as environment variable')
+
+    logger.info("Segment Event: " + json.dumps(event))  # Remove this if using in production.
+
+    # Allow Personalize region to be overriden via environment variable. Optional.
+    runtime_params = { 'service_name': 'personalize-runtime' }
+    if 'region_name' in os.environ:
+        runtime_params['region_name'] = os.environ['region_name']
+
+    personalize_runtime = boto3.client(**runtime_params)
+    personalize_events = boto3.client(service_name='personalize-events')
+
+    # Segment will invoke your function once per event type you have configured
+    # in the Personalize destination in Segment.
+
+    try:
+        if ('anonymousId' in event and
+            'properties' in event and
+            'sku' in event['properties']):
+
+            logger.info("Calling Personalize.PutEvents()")
+
+            # Function parameters for put_events call.
+            params = {
+                'trackingId': os.environ['personalize_tracking_id'],
+                'sessionId': event['anonymousId']
+            }
+
+            # If a user is signed in, we'll get a userId. Otherwise for anonymous 
+            # sessions, we will not have a userId. We still want to call put_events
+            # in both cases. Once the user identifies themsevles for the session, 
+            # subsequent events will have the userId for the same session and 
+            # Personalize will be able to connect prior anonymous to that user.
+            if event.get('userId'):
+                params['userId'] = event['userId']
+
+            # You will want to modify this part to match the event props
+            # that come from your events - Personalize needs the event identifier
+            # that was used to train the model. In this case, we're using the 
+            # product's SKU passed through Segment to represent the eventId.
+            properties = { 'itemId': event['properties']['sku'] }
```

Comment:
`itemId` is now supported in the event entity so you can move it outside of `properties` and into the event. https://docs.aws.amazon.com/personalize/latest/dg/API_UBS_PutEvents.html

FUTURE: consider including contextual metadata such as device.